jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/TreeTrie.java

```diff
@@ -276,7 +276,31 @@ private V getBest(Node<V> node, byte[] b, int offset, int len)
     @Override
     public boolean isEmpty()
     {
-        return keySet().isEmpty();
+        return !hasAnyKey(_root);
+    }
+
+    private boolean hasAnyKey(Node<V> t)
+    {
+        if (t != null)
+        {
+            if (t._key != null)
+                return true;
+
+            for (int i = 0; i < INDEX; i++)
+            {
+                if (t._nextIndex[i] != null)
+                {
+                    if (hasAnyKey(t._nextIndex[i]))
+                        return true;
+                }
+            }
+            for (int i = t._nextOther.size(); i-- > 0; )
+            {
+                if (hasAnyKey(t._nextOther.get(i)))
+                    return true;
+            }
+        }
+        return false;
```

Comment:
probably invert the boolean and rename isEmpty(Node)

Reply:
Inverted